### DISCUSSION TITLE
Dynamic NeuroDesk badge

### DISCUSSION THREAD

**Discussion by marcelzwiers** [neurodesk/neurodesk.github.io-hugo#372] [Ideas]
In my README file (which is shown as the entry point on GitHub, PyPi, and RTD) I display a few dynamic badges that inform the user about the status of the project, such as the CI status, the RTB build, but also the latest PyPi version (see [here](https://bidscoin.readthedocs.io)). The project is also included in NeuroDesk, and I would like to advertise that with a badge, showing the version of the most recent neurocontainer (and in green if it passes the tests or so). So something like this, but then dynamically:

https://img.shields.io/badge/NeuroDesk-v4.2.0-green

![image](https://github.com/NeuroDesk/neurodesk.github.io/assets/15156015/a8593557-5255-4a48-87b9-38afd85947b7)

I know it's not very important, and as you can see I can do it statically myself. It's just a rainy afternoon task, and a nice PR goody :-)

**Comment by stebo85** [comment]
Great idea :) what would be required to do this on our side? Should we create this badge automatically when we build the container and store it in the repository so you can statically link to it and it will be automatically updated on a container update?

**Comment by marcelzwiers** [comment]
I would think that when you create the neurocontainer, you could run a test and create a versioned-badge (and store it next to the container or so). The badge color could e.g. be green when the test passed, red when it didn't, and a neutral color (grey?) when there isn't a test function. Then you can serve the badge e.g. on `https://www.neurodesk.org/bidscoin/badge/?version=latest` and also display it here:
https://www.neurodesk.org/docs/overview/applications/

**Comment by marcelzwiers** [comment]
And then, when you click on it you could even serve the README of your container :-)

**Comment by marcelzwiers** [comment]
Or, instead of storing the badge, you can redirect to imshields and render the badge dynamically (which may be better for future maintenance):
https://shields.io/badges

**Comment by stebo85** [comment]
Yes, that would be nice :) 

I created an issue https://github.com/NeuroDesk/neurodesk.github.io/issues/373 

and I hope we can find someone who is keen to work on this.

Thank you
Steffen
